packages/migrate/src/__tests__/MigrateDiff.test.ts

```diff
@@ -208,6 +224,75 @@ describe('migrate diff', () => {
       await expect(result).resolves.toMatchInlineSnapshot(``)
       expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(`No difference detected.`)
     })
+
+    describe('--exit-code', () => {
+      it('should exit with code 2 when diff is not empty without --script', async () => {
+        ctx.fixture('schema-only-sqlite')
+
+        const mockExit = jest.spyOn(process, 'exit').mockImplementation()
+
+        const result = MigrateDiff.new().parse([
+          '--preview-feature',
+          '--from-schema-datamodel=./prisma/schema.prisma',
+          '--to-empty',
+          '--exit-code',
+        ])
+
+        await expect(result).resolves.toMatchInlineSnapshot(``)
+        expect(ctx.mocked['console.error'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+        expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+
+                              [-] Removed tables
+                                - Blog
+                        `)
+
+        expect(mockExit).toHaveBeenCalledTimes(1)
+        expect(mockExit).toHaveBeenCalledWith(2)
+        mockExit.mockRestore()
+      })
+
+      it('should exit with code 2 when diff is not empty with --script', async () => {
+        ctx.fixture('schema-only-sqlite')
+
+        const mockExit = jest.spyOn(process, 'exit').mockImplementation()
+
+        const result = MigrateDiff.new().parse([
+          '--preview-feature',
+          '--from-schema-datamodel=./prisma/schema.prisma',
+          '--to-empty',
+          '--script',
+          '--exit-code',
+        ])
+
+        await expect(result).resolves.toMatchInlineSnapshot(``)
+        expect(ctx.mocked['console.error'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+        expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+          -- DropTable
+          PRAGMA foreign_keys=off;
+          DROP TABLE "Blog";
+          PRAGMA foreign_keys=on;
+        `)
+
+        expect(mockExit).toHaveBeenCalledTimes(1)
+        expect(mockExit).toHaveBeenCalledWith(2)
+        mockExit.mockRestore()
+      })
+
+      it('should exit with code 0 when diff is empty with --script', async () => {
+        ctx.fixture('empty')
+
+        const result = MigrateDiff.new().parse([
+          '--preview-feature',
+          '--from-empty',
+          '--to-url=file:doesnotexists.db',
+          '--script',
+          '--exit-code',
+        ])
+
+        await expect(result).resolves.toMatchInlineSnapshot(``)
+        expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(`-- This is an empty migration.`)
```

Comment:
Is this one missing the assertion on the exit code?

Reply:
Yes because we don't test normal exit codes directly anywhere, though `resolves` is a success and `rejects` is an error.
So here a success = 0